internal/mempool/mempool.go

```diff
@@ -279,19 +279,14 @@ func (txmp *TxMempool) CheckTx(
 		return err
 	}
 
-	if txmp.recheckCursor != nil {
```

Comment:
wouldn't the "more proper" fix be to set recheck cursor to nil AFTER the recheck fails as opposed to just not checking? In this case, can't a recheck and a checkTx happen at the same time (which may cause incorrect acceptance / rejection from mempool?

Reply:
Setting recheckCurose to nil should only happen in the handleRecheckResult function after each commit is done. We can not blindly set the recheck cursor to nil because we still want to finish the pending rechecks, otherwise, the pending rechecks will be lost